dt/uni01alpha/netconfig/netconfig.yaml

```diff
@@ -0,0 +1,28 @@
+---
+apiVersion: network.openstack.org/v1beta1
+kind: NetConfig
```

Comment:
I see uni01alpha is going to get its own `netconfig` directory in so that it doesn't have the external network. That's not a problem, I'm just pointing out why this seems to have been done.

```diff
[fultonj@runcible netconfig{uni01alpha-ironic}]$ diff -u ../../../lib/networking/netconfig/netconfig.yaml netconfig.yaml 
--- ../../../lib/networking/netconfig/netconfig.yaml	2024-05-21 09:58:41.562946092 -0400
+++ netconfig.yaml	2024-06-07 13:22:50.481458107 -0400
@@ -17,11 +17,6 @@
         - _replaced_
       mtu: 1500
     - dnsDomain: _replaced_
-      name: external
-      subnets:
-        - _replaced_
-      mtu: 1500
-    - dnsDomain: _replaced_
       name: storage
       subnets:
         - _replaced_
[fultonj@runcible netconfig{uni01alpha-ironic}]$ diff -u ../../../lib/networking/netconfig/kustomization.yaml kustomization.yaml 
--- ../../../lib/networking/netconfig/kustomization.yaml	2024-05-21 09:58:41.562946092 -0400
+++ kustomization.yaml	2024-06-07 13:22:50.481458107 -0400
@@ -28,15 +28,6 @@
   - source:
       kind: ConfigMap
       name: network-values
-      fieldPath: data.external.dnsDomain
-    targets:
-      - select:
-          kind: NetConfig
-        fieldPaths:
-          - spec.networks.[name=external].dnsDomain
-  - source:
-      kind: ConfigMap
-      name: network-values
       fieldPath: data.storage.dnsDomain
     targets:
       - select:
@@ -75,15 +66,6 @@
   - source:
       kind: ConfigMap
       name: network-values
-      fieldPath: data.external.mtu
-    targets:
-      - select:
-          kind: NetConfig
-        fieldPaths:
-          - spec.networks.[name=external].mtu
-  - source:
-      kind: ConfigMap
-      name: network-values
       fieldPath: data.storage.mtu
     targets:
       - select:
@@ -122,15 +104,6 @@
   - source:
       kind: ConfigMap
       name: network-values
-      fieldPath: data.external.subnets
-    targets:
-      - select:
-          kind: NetConfig
-        fieldPaths:
-          - spec.networks.[name=external].subnets
-  - source:
-      kind: ConfigMap
-      name: network-values
       fieldPath: data.storage.subnets
     targets:
       - select:
[fultonj@runcible netconfig{uni01alpha-ironic}]$ 
```

Reply:
Yes, I was not able to remove tha External network using a patch reliably.
Would have to use a static list index :( And that means changes to in the lib would potentially break in case the list is re-ordered or something is added at the beginning/before current External network index.

We may want to look into changing the lib to have one file for each network or something so that the different DTs/VAs can pick the ones required. But let's look at that in follow ups ...